### ISSUE TITLE
[Bug]: collection ttl properties don't seem to work properly.

### ISSUE THREAD

**Issue by qhdrl12** [milvus-io/milvus#28644]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.2
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.3
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: 16core, 14GB
- GPU: 
- Others:
```

### Current Behavior

I set the ttl for a short period of time in the collection,
I'm checking to see if the added data expires properly, but it doesn't seem to be working as intended, so I'd like to ask for help.
```
collection_name = "cache_collection"
collection = Collection(name=collection_name, schema=schema, properties={"collection.ttl.seconds": 120})
```

and I checked the settings in the following way:
```
print(collection.describe().get("properties"))
> {'collection.ttl.seconds': '120'}
```

I also tried setting it with set_properties, but the result was the same.
```
collection.set_properties(properties={"collection.ttl.seconds": 120})
```

After doing a little research, it seemed like ttl comparison logic was performed when calling compact, so I manually performed compact, but the data size was the same.
```
print(collection.num_entities)
collection.compact()
print(collection.get_compaction_state())
print(collection.num_entities)
> 1723
> CompactionState
 - compaction id: 445788774451498831
 - State: Completed
 - executing plan number: 0
 - timeout plan number: 0
 - complete plan number: 0

> 1723
```

If there is anything else you need to check, please let me know.

### Expected Behavior

Set TTL 120 for the collection,
After inserting 1000 random pieces of data, there should be no data when searched 120 seconds later.

### Milvus Log

This is the milvus data coord log when performing manual compact.

```
[2023/11/22 04:38:22.365 +00:00] [INFO] [datacoord/services.go:1028] ["received manual compaction"] [traceID=0acc9db42a33a39b55d429cae204ccf6] [collectionID=445788774451444879]
[2023/11/22 04:38:22.366 +00:00] [INFO] [datacoord/services.go:1052] ["success to trigger manual compaction"] [traceID=0acc9db42a33a39b55d429cae204ccf6] [collectionID=445788774451444879] [compactionID=445788774451501203]
```

**Comment by xiaofan-luan** [comment]
1. ttl feature is more like a async cleanup. if you want to trigger cleanup you will need to run manualy compaction
2. We don't recommend to set ttl in such a short perioid, usually it should be at least a couple of days.

**Comment by xiaofan-luan** [comment]
we should have a compaction trigger which try to clean up segments with staled data.

/assign @smellthemoon

**Comment by qhdrl12** [comment]
> 1. ttl feature is more like a async cleanup. if you want to trigger cleanup you will need to run manualy compaction
I thought it would work asynchronously, so I waited for more than an hour, but TTL didn't seem to work.  
Is calling collection.compact() the correct way to perform manual compaction?

> 2. We don't recommend to set ttl in such a short perioid, usually it should be at least a couple of days.
I agree. 
Actual data will be stored for several days, and we have currently arbitrarily set a very short period of time for functional testing.

**Comment by yanliang567** [comment]
I don't think manual compaction would be 100% to cleanup the deleted/expired data in a segment. Say if there are only 1% data in a segment is deleted/expired, it would not be cleaned up by a manual compaction. 
if you want to ensure the data was expired, please try to query/get the expired data. @qhdrl12 

/unassign

**Comment by qhdrl12** [comment]
@yanliang567 
Thank you for letting me know what to check.

I created a collection, set the TTL(120), and stored 1800 pieces of data yesterday.
And I hope that when I search now, no data comes out.

But when I query, I get the following result:

```
search_param = {
    "data": [query_vector],
    "anns_field": "float_vector",
    "param": {},
    "limit": 5,
    "output_fields": ["data"]
}
res = collection.search(**search_param)
print(res)

> ['["id: 445788774450771713, distance: 0.0, entity: {\'data\': \'e43e0317-1852-45a8-9291-71e2992aeaa2\'}", "id: 445788774450771715, distance: 0.0, entity: {\'data\': \'e43e0317-1852-45a8-9291-71e2992aeaa2\'}", "id: 445788774450771717, distance: 0.0, entity: {\'data\': \'e43e0317-1852-45a8-9291-71e2992aeaa2\'}", "id: 445788774450771721, distance: 0.0, entity: {\'data\': \'e43e0317-1852-45a8-9291-71e2992aeaa2\'}", "id: 445788774450771719, distance: 0.0, entity: {\'data\': \'e43e0317-1852-45a8-9291-71e2992aeaa2\'}"]']
```

**Comment by jaime0815** [comment]
the common.retentionDuration is a global configuration for travel time,  the collection.ttl.seconds is collection level configuration.  and the common.retentionDuration has been removed in 2.3 version.

**Comment by yanliang567** [comment]
@qhdrl12 could you please provide full milvus logs for investigation? refer this [doc](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs

**Comment by qhdrl12** [comment]
@yanliang567 Sure!

I tested insert & search again with a similarly short TTL cycle and exported the log by referring to the documentation. [milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/13437064/milvus-log.tar.gz)
I don't know if the content is sufficient, but I hope it helps you check.

If you need more information, please let me know anytime.

**Comment by qhdrl12** [comment]
@smellthemoon 
Thanks for explaining in detail. As you guessed, the cause was missing flush.  :( 

After inserting and flushing the data, I checked num_entities and the desired result was obtained, and I was able to confirm that it was deleted after the ttl time.
I should have looked at the document more closely. Thanks for finding the cause 👍 

Lastly, I have a question.

I confirmed that the flush log can be viewed through proxy.
Where can I check the log when an entity is deleted due to ttl?
And should flush be called every insert, update, delete?

**Comment by smellthemoon** [comment]
> @smellthemoon Thanks for explaining in detail. As you guessed, the cause was missing flush. :(
> 
> After inserting and flushing the data, I checked num_entities and the desired result was obtained, and I was able to confirm that it was deleted after the ttl time. ![스크린샷 2023-11-29 오후 6 13 22](https://private-user-images.githubusercontent.com/15104468/286540397-4e71ad92-fde0-4c7d-bbdc-a5957db76661.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzMzcxMjAsIm5iZiI6MTcwMTMzNjgyMCwicGF0aCI6Ii8xNTEwNDQ2OC8yODY1NDAzOTctNGU3MWFkOTItZmRlMC00YzdkLWJiZGMtYTU5NTdkYjc2NjYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDA5MzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZWNhNWEyNGZlNDYyZmZiNWM0MmMwYWIwNmI4OThjNThjN2M3NTI5OWIzYTRjM2NlMTE4OTg3ZDAxZjBmZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hA0tTxUxgoe3k4rqNGVnh9PEf6eEUKmXHv-9IbXHCiE)
> 
> I should have looked at the document more closely. Thanks for finding the cause 👍
> 
> Lastly, I have a question.
> 
> I confirmed that the flush log can be viewed through proxy. Where can I check the log when an entity is deleted due to ttl? And should flush be called every insert, update, delete?
No, you can just flush once after a batch of insert/delete/upsert.

**Comment by qhdrl12** [comment]
Manual compaction failed because flush was not called(maybe), and because of this, the ttl setting did not seem to work. Flush was performed and normal operation was confirmed.
